# Actividad 1 - Estudio sobre rendimiento físico

| .    | Velocidad | <11 | 11-15 | >15 | Total |
|------|-----------|-----|-------|-----|-------|
| Edad | <=17      | 34  | 18    | 10  | 62    |
| .    | 18-29     | 25  | 23    | 21  | 69    |
| .    | >=30      | 14  | 24    | 31  | 69    |
| .    | Total     | 73  | 65    | 62  | 200   |

In [1]:
valores_observados <- data.frame(
    "menos_11" = c(34, 25, 14),
    "once_a_15" = c(18, 23, 24),
    "mayor_a_15" = c(10, 21, 31)
)
rownames(valores_observados)<-c("menor_18", "dieciocho_29", "mayor_a_30")
valores_observados

menos_11 once_a_15 mayor_a_15
menor_18     34       18        10        
dieciocho_29 25       23        21        
mayor_a_30   14       24        31

Calculamos la tabla de proporciones del total:

In [2]:
prop.table(valores_observados)

menos_11 once_a_15 mayor_a_15
menor_18     0.170    0.090     0.050     
dieciocho_29 0.125    0.115     0.105     
mayor_a_30   0.070    0.120     0.155

Para calcular la tabla de proporciones de las filas (con sumas marginales):

In [3]:
addmargins(prop.table(as.matrix(valores_observados), 1))

menos_11  once_a_15 mayor_a_15 Sum
menor_18     0.5483871 0.2903226 0.1612903  1  
dieciocho_29 0.3623188 0.3333333 0.3043478  1  
mayor_a_30   0.2028986 0.3478261 0.4492754  1  
Sum          1.1136045 0.9714820 0.9149135  3

Para calcular las proporciones de las columnas (con sumas marginales):

In [4]:
addmargins(prop.table(as.matrix(valores_observados), 2))

menos_11  once_a_15 mayor_a_15 Sum      
menor_18     0.4657534 0.2769231 0.1612903  0.9039668
dieciocho_29 0.3424658 0.3538462 0.3387097  1.0350216
mayor_a_30   0.1917808 0.3692308 0.5000000  1.0610116
Sum          1.0000000 1.0000000 1.0000000  3.0000000

Hacemos la prueba de independencia chi cuadrada de Pearson

In [5]:
# chisq es de RBase
chi2<-chisq.test(valores_observados)
chi2


	Pearson's Chi-squared test

data:  valores_observados
X-squared = 19.784, df = 4, p-value = 0.000551


El resultado anterior significa que hay _cierta asociación_ entre los valores observados.

In [6]:
chi2$observed

menos_11 once_a_15 mayor_a_15
menor_18     34       18        10        
dieciocho_29 25       23        21        
mayor_a_30   14       24        31

Hacemos la prueba de independencia de máxima verosimilitud

In [7]:
library(DescTools)
GTest(valores_observados)


	Log likelihood ratio (G-test) test of independence without correction

data:  valores_observados
G = 20.398, X-squared df = 4, p-value = 0.0004166


Extraemos los valores esperados de la tabla, mismos que valores sirven para hacer la prueba de independencia y comprobar la hipótesis nula.

In [8]:
chi2$expected

menos_11 once_a_15 mayor_a_15
menor_18     22.630   20.150    19.22     
dieciocho_29 25.185   22.425    21.39     
mayor_a_30   25.185   22.425    21.39

Extraemos los residuales _no estandarizados_ (crudos) de la prueba de independencia de Pearson.

In [9]:
chi2$residuals

menos_11    once_a_15  mayor_a_15 
menor_18      2.39011164 -0.4789619 -2.10307243
dieciocho_29 -0.03686386  0.1214232 -0.08432556
mayor_a_30   -2.22876876  0.3325939  2.07786819

Extraemos los residuales _estandarizados_ de la prueba de independencia de Pearson

In [10]:
chi2$stdres

menos_11    once_a_15  mayor_a_15
menor_18      3.61082782 -0.7018180 -3.0479311
dieciocho_29 -0.05716013  0.1826119 -0.1254336
mayor_a_30   -3.45587080  0.5001978  3.0908135

Aunque las pruebas de hipótesis de independencia estadística nos dan un criterio para considerar si hay o no una relación entre las variables que hemos tabulado el análisis de datos categóricos no se detiene allí. Si queremos saber cuáles categorías son las más divergentes –es decir, en cuáles es mayor la diferencia entre valores esperados y observados– podemos observar como se distribuyen los residuos. Los residuos no estandarizados que elaboramos anteriormente nos dan una idea de las diferencias entre valores observados y esperados, pero es difícil compararlos de manera directa ya que no son sensibles a los tamaños de fila o columna.

_Rule of Thumb for Interpreting Standardized Residuals_

A general rule of thumb for figuring out what the standardized residual means, is:

  - If the residual is less than -2, the cell’s observed frequency is less than the expected frequency.
  - Greater than 2 and the observed frequency is greater than the expected frequency.
  - If your residuals are +/-3, then it means that something extremely unusual is happening.
  - If you get +/-4, it’s something from the Twilight Zone! 
 
This makes sense if you think about the 68 95 99.7 rule: if your data is normally distributed, 95% of your data should be within 2 standard deviations from the mean. If you have something greater than that, then you’re looking at an outlier.

Obtenemos el valor crítico de Ji cuadrada de tablas.

In [11]:
r<-nrow(valores_observados)
c<-ncol(valores_observados)
grados_libertad<-(r-1)*(c-1)

In [12]:
qchisq(0.95, df=grados_libertad)

[1] 9.487729

Haciendo la prueba de hipótesis:

  - $H_0$: No existe relación entre la velocidad promedio y edad de los ciclistas

  - $H_1$: Existe relación entre la velocidad promedio y edad de los ciclistas

  De la prueba Chi cuadrada obtuvimos 19.784 y el valor crítico es 9.48772903678115. Siendo el valor calculado mucho mayor al valor crítico.

De lo anterior hay suficiente evidencia estadística para afirmar que hay _asociación_ entre la velocidad promedio y la edad de los ciclistas.

_Esto no implica causalidad, ya que no se puede afirmar que el cambio de edad cause que el ciclista tenga mayor velocidad._

Como nota de este ejercicio, vemos que las filas menor a 18 y mayor a 30 tienen una alta variación, por lo cual se necesita hacer una prueba más a profundidad para hacer las afirmaciones anteriores.